Classes/Finishers/EmailFinisher.php

```diff
@@ -273,6 +274,9 @@ protected function addAttachments(Email $mail)
         $formValues = $this->finisherContext->getFormValues();
         if ($this->parseOption('attachAllPersistentResources')) {
             foreach ($formValues as $formValue) {
+                if ($formValue instanceof Image) {
+                    $formValue = $formValue->getResource();
+                }
```

Comment:
This adds a dependency to the `neos/media` package, doesn't it?

Reply:
Damn, you are right. Was not aware that `Neos.Form` just depends on `Neos/Flow`.

What do you think about replacing it with:
```php
if (method_exists($formValue, 'getResource')) {
    $formValue = $formValue->getResource();
}
```

feels quite hacky ... Maybe `Neos.Form.Builder` should somehow extend this finisher? Would require a larger concept 🤔

Reply:
Actually it's not a hard dependency. But yeah... quesdtion is, if we want to put that into the form package.
https://3v4l.org/4Z7ll#vnull

Reply:
@bwaidelich I think there is already a problem in this repository! The [ImageUpload](https://github.com/neos/form/blob/main/Classes/FormElements/ImageUpload.php) there is already a dependency to `neos/media`. Also in the `RenderValuesViewHelper` and other classes and viewhelpers. So this package already does not work without `neos/media`. Think the proper fix would be to add `neos/media` as dependency

Reply:
Ihhh.... But we can't make neos/media a dependency. ATM you can use just partial functionality of the package, which does not use the media package.

If we want to keep that, we should at least check for `ResourceBasedInterface` instead of image. This would cover all assets.

Reply:
Can we merge this PR and create a new issue  to discuss the dependency issues?

Reply:
As we already have soft constraints to media package since ages, I'm fine with this. @bwaidelich WDYT?